packages/client/src/index.ts

```diff
@@ -39,6 +42,63 @@ export class Client extends AutoGeneratedClient {
     super(undefined, clientConfig.apiUrl, axiosClient)
 
     this.config = clientConfig
+    this.axiosClient = axiosClient
+  }
+
+  /**
+   * Creates and uploads a new file in a single step. Returns an object containing the file metadata and the URL to retrieve the file.
+   */
+  public createAndUploadFile = async ({
+    name,
+    data,
+    index,
+    tags,
+    contentType,
+    contentLength,
+    accessPolicies,
+  }: CreateFileProps & { data: Buffer; contentLength?: number }): Promise<GetFileResponse> => {
+    const headers = {
+      ...this.config.headers,
+      'content-length': contentLength,
+    }
+
+    const requestBody: CreateFileBody = {
+      name,
+      tags,
+      index,
+      accessPolicies,
+      contentType,
+    }
+
+    const response = await this.axiosClient
+      .post('/v1/files', requestBody, { headers, baseURL: this.config.apiUrl })
+      .catch((err) => {
+        throw getError(err)
+      })
+
+    const { file } = response.data as CreateFileResponse
+
+    try {
+      await axios.put(file.uploadUrl, data, {
+        maxBodyLength: Infinity,
+      })
+    } catch (err: any) {
+      throw new FileUploadError(`Failed to upload file: ${err.message}`, <AxiosError>err, file)
+    }
+
+    // Calling this endpoint triggers the processing of the upload on the API side and allows to return the retrieval URL and the file with the correct status.
```

Comment:
This is internal behaviour that should not be exposed in the client

Reply:
Done in https://github.com/botpress/botpress/pull/13076/commits/f422ddc0ace46411c0e4bf643a20243087b319bc